docs/ovs-pipeline.md

```diff
@@ -0,0 +1,604 @@
+to drop traffic originating from our 2 Pods (10.10.1.2 and 10.10.1.3), which is
+confirmed by dumping the flows:
+```
+1. table=60, priority=200,ip,nw_src=10.10.1.2 actions=drop
+2. table=60, priority=200,ip,nw_src=10.10.1.3 actions=drop
+3. table=60, priority=0 actions=resubmit(,70)
+```
+
+The table-miss flow entry, which is used for non-isolated Pods, resubmits
+traffic to the next table ([L3ForwardingTable]).
+
+### L3ForwardingTable (70)
+
+This is the L3 routing table. It implements the following functionality:
+
+ * Tunnelled traffic coming-in from a peer Node and destined to a local Pod is
+   directly forwarded to the Pod. This requires setting the source MAC to the
+   MAC of the local gateway interface and setting the destination MAC to the
+   Pod's MAC address. Such traffic is identified by matching on the packet's
+   destination MAC address (should be set to the Global Virtual MAC for all
+   tunnelled traffic) and its destination IP address (should match the IP
+   address of a local Pod). We therefore install one flow for each Pod created
+   locally on the Node. For example:
+```
+table=70, priority=200,ip,dl_dst=aa:bb:cc:dd:ee:ff,nw_dst=10.10.0.2 actions=mod_dl_src:e2:e5:a4:9b:1c:b1,mod_dl_dst:12:9e:a6:47:d0:70,dec_ttl,resubmit(,80)
+```
+
+ * All tunnelled traffic destined to the local gateway (i.e. for which the
+   destination IP matches the local gateway's IP) is forwarded to the gateway
+   port by rewriting the destination MAC (from the Global Virtual MAC to the
+   local gateway's MAC).
+```
+table=70, priority=200,ip,dl_dst=aa:bb:cc:dd:ee:ff,nw_dst=10.10.0.1 actions=mod_dl_dst:e2:e5:a4:9b:1c:b1,resubmit(,80)
+```
+
+ * All traffic destined to a remote Pod is forwarded through the appropriate
+   tunnel. This means that we install one flow for each peer Node, each one
+   matching the destination IP address of the packet against the Pod subnet for
+   the Node. In case of a match the source MAC is set to the local gateway MAC,
+   the destination MAC is set to the Global Virtual MAC and we set the OF
+   `tun_dst` field to the appropriate value (i.e. the IP address of the remote
+   gateway). Traffic is then resubmitted to [ConntrackCommitTable], thus
+   skipping [L2ForwardingCalcTable] and the ingress policy rules tables, which
+   are not relevant for traffic destined to a tunnel (the destination port is
+   the tunnel port and the ingress policy rules will be enforced at the
+   destination Node). For a given peer Node, the flow may look like this:
+```
+table=70, priority=200,ip,nw_dst=10.10.1.0/24 actions=dec_ttl,mod_dl_src:e2:e5:a4:9b:1c:b1,mod_dl_dst:aa:bb:cc:dd:ee:ff,load:0x1->NXM_NX_REG1[],load:0x1->NXM_NX_REG0[16],load:0xc0a84d65->NXM_NX_TUN_IPV4_DST[],resubmit(,105)
+```
+
+All this traffic will then be resubmitted to the next table
+([L2ForwardingCalcTable]).
+
+If none of the flows described above are hit, traffic
+goes directly to [L2ForwardingCalcTable]. This is the case for external traffic,
```

Comment:
This is required for both externel traffic and local traffic(local Pod to local Pod)

Reply:
Fixed.